src/main/java/net/neoforged/neoforge/transfer/fluids/templates/AttachmentFluidStorage.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.fluids.templates;
+
+import net.neoforged.neoforge.attachment.AttachmentHolder;
+import net.neoforged.neoforge.attachment.AttachmentType;
+import net.neoforged.neoforge.fluids.SimpleFluidContent;
+import net.neoforged.neoforge.transfer.ResourceStack;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.fluids.FluidResource;
+import net.neoforged.neoforge.transfer.handlers.ISingleResourceHandler;
+
+import java.util.function.Predicate;
+
+public class AttachmentFluidStorage implements ISingleResourceHandler<FluidResource> {
+    private final AttachmentHolder holder;
+    private final AttachmentType<ResourceStack<FluidResource>> attachmentType;
+    private final int limit;
+    private Predicate<FluidResource> validator = r -> true;
+
+    public AttachmentFluidStorage(AttachmentHolder holder, AttachmentType<ResourceStack<FluidResource>> attachmentType, int limit) {
+        this.holder = holder;
+        this.attachmentType = attachmentType;
+        this.limit = limit;
+    }
+
+    public AttachmentFluidStorage setValidator(Predicate<FluidResource> validator) {
+        this.validator = validator;
+        return this;
+    }
+
+    @Override
+    public FluidResource getResource() {
+        return holder.getData(attachmentType).resource();
+    }
+
+    @Override
+    public int getAmount() {
+        return holder.getData(attachmentType).amount();
+    }
+
+    @Override
+    public int getCapacity(FluidResource resource) {
+        return limit;
+    }
+
+    @Override
+    public int getCapacity() {
+        return 0;
```

Comment:
Mentioned this on discord, but this should either call `limit` or potentially `getCapacity(0)`